allennlp/commands/evaluate.py

```diff
@@ -96,7 +97,13 @@ def evaluate(model: Model,
         for batch in generator_tqdm:
             model(**batch)
             metrics = model.get_metrics()
-            description = ', '.join(["%s: %.2f" % (name, value) for name, value in metrics.items()]) + " ||"
+            if (not _warned_tqdm_ignores_underscores and
```

Comment:
instead of creating this variable, why not just check for underscores and warn before the `for batch in` loop?